narwhals/_pandas_like/dataframe.py

```diff
@@ -838,12 +837,11 @@ def to_numpy(self, dtype: Any = None, *, copy: bool | None = None) -> _2DArray:
                 return self.native.to_numpy(dtype=dtype, copy=copy)
             return self.native.to_numpy(copy=copy)
 
-        dtypes = import_dtypes_module(self._version)
-
+        dtype_datetime = self._version.dtypes.Datetime
         to_convert = [
             key
             for key, val in self.schema.items()
-            if val == dtypes.Datetime and val.time_zone is not None  # type: ignore[attr-defined]
+            if isinstance(val, dtype_datetime) and val.time_zone is not None
```

Comment:
I think this is the only difference that might cause an ~issue~ difference downstream:

```py
from narwhals.dtypes import Datetime as MainDatetime
from narwhals.stable.v1 import Datetime as V1Datetime

MainDatetime() == V1Datetime  # True
isinstance(MainDatetime(), V1Datetime)  # False
```

(but I didn't check further)

Reply:
1. Yes I agree that this might not be expected behavior :| 
2. You are right, it should not be reachable! Then I am as lost as you are/were on the ci failure that happened 🤔